### DISCUSSION TITLE
Create a legacy connection for a data source to a model

### DISCUSSION THREAD

**Discussion by rhains** [TabularEditor/TabularEditor3#985] [🙋♀️TE3 Q&A]
I am migrating Tabular models from SSAS to Power BI premium service. 
I run into a Tabular model which one of the data sources connects to another tabular model.
I am looking help on how to create a Legacy data source connection string using MSOLAP provider that will allow me to connect to the model without using XMLA. 
Anyone can help with this request?

Richard

**Comment by otykier** [comment]
Hi Richard,
I believe it should be possible to make a legacy connection to another tabular model using OLE DB and the MSOLAP provider. In Tabular Editor 3, the steps are the following:-

1. Go to the **Model** menu, choose **Import tables...**
2. Choose the **Use provider (aka. Legacy) data source**
3. Give the source a name and choose **OLE DB**
4. In the Data Link Properties dialog, choose **Microsoft OLE DB Provider for Analysis Services** (this is also known as MSOLAP).

![image](https://github.com/TabularEditor/TabularEditor3/assets/8976200/c123eef7-361f-4201-805f-47706edb33d4)

5. Click **Next**
6. Specify the server name as "Data Source", choose **Use Windows NT Integrated security** for now, and select the model in the **initial catalog** dropdown.

![image](https://github.com/TabularEditor/TabularEditor3/assets/8976200/5608f62a-f7c5-4473-9062-9afee9e0d967)

7. Click **OK**
8. You can either choose from a list of tables or write an MSOLAP SQL/MDX statement.
9. If you choose from a list of available tables, be aware that the preview currently does not work, so we recommend setting **Data preview** to "Schema only" _before_ selecting a table.

<img width="593" alt="image" src="https://github.com/TabularEditor/TabularEditor3/assets/8976200/7ce63d97-38b1-4e6d-940e-ddd33150f293">

10. After importing, your table and partition should look like below. You can use TE3's batch rename feature to clean up the names.

![image](https://github.com/TabularEditor/TabularEditor3/assets/8976200/ad06339f-1497-4e34-a4f6-8a80aa417cbc)

Note that for refreshes to work once the model is published to the Power BI service, you will have to set up an [on-premises gateway](https://learn.microsoft.com/en-us/power-bi/connect-data/service-gateway-onprem).

**Reply by rhains** [reply]
Hi Otykier, I followed your steps and I make it work, after installing x64_16.0.5351.0_SQL_AS_OLEDB.msi which I did not have in my pc. However, when I deploy the model in Power BI Premium I go to Datasets Setting and I do not see the Gateway connection settings.
That indicates me the connection string might be wrong. 
Any suggestions?
